tests/suite/test_dos.py

```diff
@@ -0,0 +1,192 @@
+
+    request.addfinalizer(fin)
+
+    return DosSetup(req_url)
+
+
+@pytest.mark.smoke
+@pytest.mark.dos
+@pytest.mark.parametrize(
+    "crd_ingress_controller_with_dos",
+    [
+        {
+            "extra_args": [
+                f"-enable-custom-resources",
+                f"-enable-app-protect-dos",
+            ]
+        }
+    ],
+    indirect=["crd_ingress_controller_with_dos"],
+)
+class TestDos:
+    def test_ap_nginx_config_entries(
+        self, kube_apis, crd_ingress_controller_with_dos, dos_setup, test_namespace
+    ):
+        """
+        Test to verify Dos annotations in nginx config
+        """
+        conf_annotations = [
+            f"app_protect_dos_enable on;",
+            f"app_protect_dos_security_log_enable on;",
+            f"app_protect_dos_monitor \"dos.example.com\";",
+            f"app_protect_dos_name \"dos.example.com\";",
+        ]
+
+        create_ingress_with_dos_annotations(
+            kube_apis, src_ing_yaml, test_namespace, "True", "True", "127.0.0.1:514"
+        )
+
+        ingress_host = get_first_ingress_host_from_yaml(src_ing_yaml)
+        ensure_response_from_backend(dos_setup.req_url, ingress_host, check404=True)
+
+        pod_name = get_first_pod_name(kube_apis.v1, "nginx-ingress")
+
+        result_conf = get_ingress_nginx_template_conf(
+            kube_apis.v1, test_namespace, "dos-ingress", pod_name, "nginx-ingress"
+        )
+        delete_items_from_yaml(kube_apis, src_ing_yaml, test_namespace)
+
+        for _ in conf_annotations:
+            assert _ in result_conf
+
+    def test_dos_sec_logs_on(
+        self,
+        kube_apis,
+        ingress_controller_prerequisites,
+        crd_ingress_controller_with_dos,
+        dos_setup,
+        test_namespace,
+    ):
+        """
+        Test corresponding log entries with correct policy (includes setting up a syslog server as defined in syslog.yaml)
+        """
+        src_syslog_yaml = f"{TEST_DATA}/dos/dos-syslog.yaml"
+        log_loc = f"/var/log/messages"
+
+        create_items_from_yaml(kube_apis, src_syslog_yaml, test_namespace)
+
+        syslog_ep = get_service_endpoint(kube_apis, "syslog-svc", test_namespace)
+
+        # items[-1] because syslog pod is last one to spin-up
+        syslog_pod = kube_apis.v1.list_namespaced_pod(test_namespace).items[-1].metadata.name
+
+        create_ingress_with_dos_annotations(
+            kube_apis, src_ing_yaml, test_namespace, "True", "True", f"{syslog_ep}:514"
+        )
+        ingress_host = get_first_ingress_host_from_yaml(src_ing_yaml)
+
+        print("--------- Run test while DOS module is enabled with correct policy ---------")
+
+        ensure_response_from_backend(dos_setup.req_url, ingress_host, check404=True)
+
+        print("----------------------- Send request ----------------------")
+        response_block = requests.get(
+            dos_setup.req_url, headers={"host": "dos.example.com"}, verify=False
```

Comment:
what is the difference between this request and the request in line 181? they look the same. why do we need two?

Reply:
well spotted, de-duped